### PR TITLE
refactor(core): change SidebarUserSettings to accept a list of providerSettings

### PR DESCRIPTION
**Pull request by Fox32**
This allows to configure custom provider settings without the need to modify the core package.

Had some issues with modifying the template app as the linter is complaining.

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/core/src/layout/Sidebar/UserSettings.tsx [hunk 1]
```diff
@@ -53,41 +40,8 @@ export function SidebarUserSettings() {
...
-          />
-        )}
-        {providers.includes('oauth2') && (
-          <OIDCProviderSettings
-            title="YourOrg"
-            apiRef={oauth2ApiRef}
-            icon={Star}
-          />
-        )}
+        {providerSettings}
```
Comment (Fox32) [inline packages/core/src/layout/Sidebar/UserSettings.tsx:43]: As an alternativ implementation we could check here for `null` and use `<DefaultProviderSettings />` instead. Than this change doesn't require any modification to the template or app.

**Comment by Fox32** [comment]
Oh, Windows CI is back 😕 😆

**Comment by Fox32** [comment]
@freben I'm not sure what's wrong with the Windows build here 😕

**Comment by freben** [comment]
Hey! The issue with windows builds on github has been resolved. You may want to update this PR on top of the latest master.

For reference: https://github.com/spotify/backstage/pull/2152

**Comment by Fox32** [comment]
Still fails 😕

**Comment by freben** [comment]
Cool, this is an entirely different thing though where the test probably catches something real, let me check.

**Review by freben — APPROVED** [review]
Honestly all of this should be in a page of its own instead of the sidebar eventually. :) But not as part of this PR!

**Pull request merged**